src/openloco/windows/station_list.cpp

```diff
@@ -17,36 +37,739 @@ namespace openloco::ui::windows::station_list
+                input::cancel_tool();
+
+            // Still active?
+            window = WindowManager::bringToFront(WindowType::stationList, companyId);
+        }
+
+        if (window == nullptr)
+        {
+            // 0x00491010
+            window = WindowManager::createWindow(
+                WindowType::stationList,
+                window_size,
+                window_flags::flag_11,
+                &_events);
+
+            window->number = companyId;
+            window->owner = companyId;
+            window->current_tab = 0;
+            window->frame_no = 0;
+            window->var_844 = 0;
+            window->var_83C = 0;
+            window->row_hover = -1;
+
+            sub_4910E8(window);
+
+            window->min_width = min_dimensions.width;
+            window->min_height = min_dimensions.height;
+            window->max_width = max_dimensions.width;
+            window->max_height = max_dimensions.height;
+            window->flags |= window_flags::resizable;
+
+            auto interface = objectmgr::get<interface_skin_object>();
+            window->colours[1] = interface->colour_0A;
+        }
+
+        // TODO: only needs to be called once.
+        init_events();
+
+        window->current_tab = 0;
+        window->invalidate();
+
+        window->widgets = _widgets;
+        window->enabled_widgets = (1 << close_button) | (1 << tab_all_stations) | (1 << tab_rail_stations) | (1 << tab_road_stations) | (1 << tab_airports) | (1 << tab_ship_ports) | (1 << company_select) | (1 << sort_name) | (1 << sort_status) | (1 << sort_total_waiting) | (1 << sort_accepts) | (1 << scrollview);
+
+        window->activated_widgets = 0;
+        window->holdable_widgets = 0;
+
+        window->call_on_resize();
+        window->call_prepare_draw();
+        window->init_scroll_widgets();
+
+        return window;
+    }
+
     window* open(uint16_t companyId, uint8_t type)
     {
         if (type > 4)
             throw std::domain_error("Unexpected station type");
 
+        window* station_list = open(companyId);
+        widx target = tabInformationByType[type].widgetIndex;
+        station_list->call_on_mouse_up(target);
+
+        return station_list;
+    }
+
+    // 0x004919A4
+    static ui::cursor_id cursor(window* window, int16_t widgetIdx, int16_t xPos, int16_t yPos, ui::cursor_id fallback)
+    {
+        if (widgetIdx != widx::scrollview)
+            return fallback;
+
+        uint16_t currentIndex = yPos / rowHeight;
+        if (currentIndex < window->var_83C && window->row_info[currentIndex] != 0xFFFF)
+            return cursor_id::hand_pointer;
+
+        return fallback;
+    }
+
+    // 0x0049196F
+    static void event_08(window* window)
+    {
+        window->flags |= window_flags::flag_14;
+    }
+
+    // 0x00491977
+    static void event_09(window* window)
+    {
+        if ((window->flags & window_flags::flag_14) == 0)
+            return;
+
+        if (window->row_hover == 0xFFFF)
```

Comment:
We should use either 0xFFFF or -1 consistently. I'm in favour of the latter

Reply:
Agreed. We should change row_hover to an `int16_t`, then, to prevent casts.